docs/develop/gettingstarted.rst

```diff
@@ -0,0 +1,310 @@
+
+.. code-block:: console
+
+   $ cd micropython
+   $ git remote add upstream https://github.com/micropython/micropython.git
+
+It is common to configure ``upstream`` and ``origin`` on a forked repository
+to assist with sharing code changes. You can maintain your own mapping but
+it is recommended that ``origin`` maps to your fork and ``upstream`` to the main
+MicroPython repository:
+
+.. code-block:: console
+
+   $ cd micropython
+   $ git remote add upstream https://github.com/micropython/micropython.git
+   $ git remote set-url origin https://github.com/<your-user-name>/micropython.git
+
+After the above configuration, your setup should be similar to this:
+
+.. code-block:: console
+   
+   $ git remote -v
+   origin	https://github.com/<your-user-name>/micropython.git (fetch)
+   origin	https://github.com/<your-user-name>/micropython.git (push)
+   upstream	https://github.com/micropython/micropython.git (fetch)
+   upstream	https://github.com/micropython/micropython.git (push)
+
+You should now have a copy of the source code. By default you are pointing
+to the master branch. To prepare for further development, it is recommended
+to work on a development branch.
+
+.. code-block:: console
+
+    $ git checkout -b dev-branch
+
+You can give it any name. You will have to compile MicroPython whenever you change 
+to a different branch.
+
+Compile and build the Code
+--------------------------
+
+For development, you should compile MicroPython for a specific platform which corresponds
+to a port. Start by installing the required dependencies. Then build the MicroPython 
+cross-compiler before you can successfully compile and build.
+
+Required dependencies
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Install the required dependencies for linux:
+
+.. code-block:: console
+
+   $ sudo apt-get install build-essential libffi-dev git pkg-config
+
+For the stm32 port, the ARM cross-compiler is required:
+
+.. code-block:: console
+
+   $ sudo apt-get install arm-none-eabi-gcc arm-none-eabi-binutils arm-none-eabi-newlib
+
+See the `ARM GCC
+toolchain <https://developer.arm.com/tools-and-software/open-source-software/developer-tools/gnu-toolchain/gnu-rm>`_
+for the latest details.
+
+Python 3 is also recommended:
+
+.. code-block:: console
+
+   $ python3 -v
+   Python 3.5.0 (default, Jul 17 2020, 14:04:10) 
+   [GCC 5.4.0 20160609] on linux
+   Type "help", "copyright", "credits" or "license" for more information.
+   >>> 
+
+All supported ports have different dependency requirements, see their respective
+`readme file <https://github.com/micropython/micropython/tree/master/ports>`_.
+
+Build the cross-compiler
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Almost all ports require building ``mpy-cross`` first to perform pre-compilation
+of scripts that will be included in the port firmware:
+
+.. code-block:: console
+
+   $ cd mpy-cross
+   $ make
+
+Also, ``mpy-cross`` must be built for the host architecture and not the target architecture.
```

Comment:
Can I move this to a note?

Reply:
I remember a review that specifically requested for this statement to be added so I added it in a separate note for now to avoid confusion. But maybe the wording may still be confusing.